src/rendering/renderers/shared/buffer/Buffer.ts

```diff
@@ -173,24 +194,56 @@ export class Buffer extends EventEmitter<{
 
     set data(value: TypedArray)
     {
-        if (this._data !== value)
+        this.setData(value, value.length);
+    }
+
+    /**
+     * Sets the data in the buffer to the given value. This will immediately update the buffer on the GPU.
+     * If you only want to update a subset of the buffer, you can pass in the size of the data.
+     * @param value - the data to set
+     * @param size - the size of the data in bytes
+     */
+    public setData(value: TypedArray, size?: number)
```

Comment:
Seems weird to have 2 api's for the same thing but with an optional arg

```js
buffer.data = typedArray
buffer.setData(typedArray)
```

we should either remove the setter of `buffer.data`, or change the name of `setData` to something like `setDataWithSize()` and not make the `size` arg optional

Reply:
maybe a different name might be better, this function does two things
- sets the data
- sets the size of the data to use.

eg:
```buffer.setData(typedArray, 2)```

Reply:
I think Sean's suggestion of renaming the function to `setDataWithSize` and making all arguments mandatory might be better, there is a slight chance this may confuse people.

The renamed and mandatory version has a much clearer purpose, with no ambiguity as to which api to use if you just had the data alone.